app/src/main/java/com/google/droidjump/achievements_data/AchievementsAdapter.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.droidjump.achievements_data;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+import androidx.annotation.NonNull;
+import androidx.fragment.app.FragmentActivity;
+import androidx.recyclerview.widget.RecyclerView;
+import com.google.android.gms.common.images.ImageManager;
+import com.google.android.gms.games.achievement.Achievement;
+import com.google.android.gms.games.achievement.AchievementBuffer;
+import com.google.droidjump.R;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Displays a list of achievements in RecyclerView.
+ **/
+public class AchievementsAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+    private AchievementBuffer achievementBuffer;
+    private FragmentActivity activity;
+    private List<RecyclerViewItem> items;
+    private List<String> sectionNames;
+
+    public AchievementsAdapter(AchievementBuffer achievementBuffer, FragmentActivity activity) {
+        this.achievementBuffer = achievementBuffer;
+        this.activity = activity;
+        fillViewItems();
+    }
+
+    private void fillViewItems() {
+        items = new ArrayList<>();
+        sectionNames = new ArrayList<>();
+        for (int i = 0; i < achievementBuffer.getCount(); i++) {
+            if (i == 0 && achievementBuffer.get(i).getState() == Achievement.STATE_UNLOCKED) {
+                items.add(new RecyclerViewItem(ItemType.SECTION_NAME, sectionNames.size()));
+                sectionNames.add(String.valueOf(R.string.unlocked_section_name));
+            } else if (i == 0 || !(achievementBuffer.get(i).getState() == Achievement.STATE_UNLOCKED) && achievementBuffer.get(i - 1).getState() == Achievement.STATE_UNLOCKED) {
```

Comment:
Would also suggest to extract 
```
achievementBuffer.get(i - 1).getState() == Achievement.STATE_UNLOCKED
```
into separate variable to make the if clause readable - currently is it too long.